R/tag_examplesShinylive.R

```diff
@@ -0,0 +1,202 @@
+#' Custom `examplesShinylive` tag.
+#'
+#' This function generates a new "Examples in Shinylive" section in the documentation. This section contains URL to
+#' the application in Shinylive as well as an iframe with the application.
+#' If no code is provided then the code is taken from the following `@examples` or `@examplesIf` tag.
+#'
+#' The application code must be executable inside Shinylive. If the application code includes functions from your
+#' package, you must add `library(<your package>)` beforehand. For more information, refer to the Decoration section
+#' on how to use and decorate existing examples.
+#'
+#' Note: All the packages used in the application code need to be installable in WebR.
+#' See [this article](https://docs.r-wasm.org/webr/latest/packages.html) for more details.
+#'
+#' @section Decoration:
+#'
+#' To avoid repetition between the `@examplesShinylive` and `@examples` sections, there are special string literals
+#' that provide access to the other tags from within the content of `@examplesShinylive`.
+#' These literals should be used as expressions embraced with `{{ }}`, which are then interpolated using
+#' `glue::glue_data(..., .open = "{{", .close = "}}")`.
+#'
+#' The following keywords are available:
+#' * `"{{ tags_examples }}"` - a list of `@examples` or `@examplesIf` tags
+#' * `"{{ examples }}"` - a list of "raw" elements from `tags_examples` list elements
+#' * `"{{ next_example }}"` - "raw" element of the next example
+#' * `"{{ prev_example }}"` - "raw" element of the previous example
+#'
+#' This allows you to access and decorate existing example code to create executable application code for Shinylive.
+#' Refer to the examples section for possible use cases.
+#'
+#' @name tag-examplesShinylive
+#'
+#' @usage
+#' #' @examplesShinylive${1:# example code (optional)}
+#'
+#' @examples
+#' # As a part of documentation:
+#'
+#' # basic example:
+#' #' (docs)
+#' #' @examplesShinylive
+#' #' @examples
+#' #' (example code)
+#'
+#' # using keywords:
+#' #' (docs)
+#' #' @examplesShinylive
+#' #' foo <- 1
+#' #' {{ next_example }}
+#' #' bar <- 2
+#' #' @examples
+#' #' (your example code)
+#'
+#' # A typical example would be:
+#' #' (docs)
+#' #' @examplesShinylive
+#' #' library(<your package>)
+#' #' interactive <- function() TRUE
```

Comment:
What's the purpose of this line?
`interactive <- function() TRUE`

Is this something related to shinylive.io?

Reply:
This overwrites `interactive()` which returns FALSE when is not interactive. It forces R process to open run shiny apps.

Reply:
In our packages we usually do the following:
```
app <- ...
if (interactive()) {
  shinyApp(...)
}
```

Examples in [tmg](https://github.com/search?q=repo%3Ainsightsengineering%2Fteal.modules.general+%22%23%27+if+%28interactive%28%29%29+%7B%22&type=code) and [tmc](https://github.com/search?q=repo%3Ainsightsengineering%2Fteal.modules.clinical+%22%23%27+if+%28interactive%28%29%29+%7B%22&type=code)

As @gogonzo explained - we overwrite that function so as to make sure it always return TRUE and opens the app

This is only our own convention so I have not _hardcoded_ this (so that it is add no matter what) but rather present usage with that line as an additional input